crates/server/src/server/server_future.rs

```diff
@@ -821,6 +888,50 @@ pub(crate) async fn handle_request<R: ResponseHandler, T: RequestHandler>(
     }
 }
 
+/// Checks if the IP address is safe for returning messages
+///
+/// Examples of unsafe addresses are any with a port of `0`
+///
+/// # Returns
+///
+/// Error if the address should not be used for returned requests
+fn sanitize_src_address(src: SocketAddr) -> Result<(), String> {
+    // currently checks that the src address aren't either the undefined IPv4 or IPv6 address, and not port 0.
+    if src.port() == 0 {
+        return Err(format!("cannot respond to src on port 0: {}", src));
+    }
+
+    fn verify_v4(src: Ipv4Addr) -> Result<(), String> {
+        if src.is_unspecified() {
+            return Err(format!("cannot respond to unspecified v4 addr: {}", src));
+        }
+
+        if src.is_broadcast() {
+            return Err(format!("cannot respond to broadcast v4 addr: {}", src));
+        }
+
+        // TODO: add check for is_reserved when that stabilizes
+
+        Ok(())
+    }
+
+    fn verify_v6(src: Ipv6Addr) -> Result<(), String> {
+        if src.is_unspecified() {
+            return Err(format!("cannot respond to unspecified v6 addr: {}", src));
+        }
+
+        Ok(())
+    }
+
+    // currently checks that the src address aren't either the undefined IPv4 or IPv6 address, and not port 0.
+    match src.ip() {
+        IpAddr::V4(v4) => verify_v4(v4)?,
+        IpAddr::V6(v6) => verify_v6(v6)?,
+    }
+
+    Ok(())
```

Comment:
Minor knit pick: We could remove this `Ok(())` if there matched arms are not checked with `?`

Reply:
I’ll make that change. Good suggestion.